### PR TITLE
Add sd

### PR DESCRIPTION
**Pull request by ganmacs**
ref https://github.com/fluent/fluentd/pull/2541

### CODE REVIEW

File: plugins/service_discovery/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+</source>
+```
+
+
+## List of Built-in Service Discovery
+
+-   [static](/plugins/service_discovery/static.md)
+-   [file](/plugins/service_discovery/file.md)
+
+## List of Core Input Plugins with Parser support
```
Comment (repeatedly) [inline plugins/service_discovery/README.md]: List of Core Output Plugins with Service Discovery support?

File: plugins/service_discovery/file.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+### @type
+
+The value must be `file`.
+
+### path
+
+| type   | default                 | version |
+|:-------|:------------------------|:--------|
+| string | `'/etc/fluent/sd.yaml'` | 0.17.0  |
```
Comment (repeatedly) [inline plugins/service_discovery/file.md]: 1.8.0, not 0.17.0. Other parameters are same.

File: plugins/service_discovery/static.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+This is the exmple with out_forward.
+It define the list of targets. it is similer to the server directive in out_forward.
+
+```
+<source>
+  @type forward
+
+  <service_discovery>
+    @type static
+    host 127.0.0.1
```
Comment (repeatedly) [inline plugins/service_discovery/static.md]: should wrap `<service>`?

File: plugins/service_discovery/file.md (hunk 2 above)
Comment (ganmacs) [reply plugins/service_discovery/file.md]: fixed in https://github.com/fluent/fluentd-docs-gitbook/pull/125

File: plugins/service_discovery/README.md (hunk 1 above)
Comment (ganmacs) [reply plugins/service_discovery/README.md]: fixed in https://github.com/fluent/fluentd-docs-gitbook/pull/125

Comment (ganmacs) [reply plugins/service_discovery/README.md]: fixed in https://github.com/fluent/fluentd-docs-gitbook/pull/125

File: plugins/service_discovery/file.md (hunk 2 above)
Comment (ganmacs) [reply plugins/service_discovery/file.md]: in https://github.com/fluent/fluentd-docs-gitbook/pull/125

File: plugins/service_discovery/static.md (hunk 3 above)
Comment (ganmacs) [reply plugins/service_discovery/static.md]: You mean as below?

```
<service_discovery>
   @type static
   <server>
     host 127.0.0.1
   </server>
</service_discovery>
```
I don't think so. see https://github.com/fluent/fluentd/pull/2541#issue-304197753 description.

File: plugins/service_discovery/file.md (hunk 2 above)
Comment (ganmacs) [reply plugins/service_discovery/file.md]: ah, 1.8.0 is correct... fixed

**Comment by repeatedly** [comment]
@ganmacs Could you also add service discovery plugin helper document? It is important for plugin developers.

**Comment by ganmacs** [comment]
Good catch. I forgot that. will do.

**Pull request merged**